### ISSUE TITLE
Can't find `sysimg.jl` with out-of-tree builds of julia 0.6

### ISSUE THREAD

**Issue by tkluck** [timholy/Revise.jl#77]
`using Revise` fails when running Julia from an out-of-tree build with the following error:

```
julia> using Revise
INFO: Precompiling module Revise.
ERROR: LoadError: SystemError: realpath: No such file or directory
Stacktrace:
 [1] #systemerror#44 at ./error.jl:64 [inlined]
 [2] systemerror(::Symbol, ::Bool) at ./error.jl:64
 [3] realpath(::String) at ./path.jl:311
 [4] include_from_node1(::String) at ./loading.jl:576
 [5] include(::String) at ./sysimg.jl:14
 [6] anonymous at ./<missing>:2
while loading /home/tkluck/.julia/v0.6/Revise/src/Revise.jl, in expression starting on line 669
ERROR: Failed to precompile Revise to /home/tkluck/.julia/lib/v0.6/Revise.ji.
Stacktrace:
 [1] compilecache(::String) at ./loading.jl:710
 [2] _require(::Symbol) at ./loading.jl:497
 [3] require(::Symbol) at ./loading.jl:405
```

The error is coming from this line:
```julia
const sysimg_path =  # where `baremodule Base` is defined
    realpath(joinpath(JULIA_HOME, Base.DATAROOTDIR, "julia", "base", "sysimg.jl"))
```
since there is no file `sysimg.jl` at that place. Adding a symlink as follows
```sh
~/src/julia/build-stable/base$ ln -s ../../base/sysimg.jl
```
solves the issue.

For reproducing such an out-of-tree build, one can use (in julia's top-level source directory):
```sh
$ make O=build-out-of-tree configure
$ make -C build-out-of-tree
```

I'm not comfortable speculating about whether this should be solved in `Revise.jl` or in the out-of-tree build scripts, but I'm happy to provide a patch given a few pointers. In any case, we could mitigate the issue in `Revise` by ignoring this error and not tracking `Base` in this case.

**Comment by timholy** [comment]
Thanks for the nice report. You haven't tested 0.7 by any chance? A lot of the file-handling has been changed for Julia 0.7 (and it's much nicer).

**Comment by tkluck** [comment]
Indeed, this seems to work under the current `master` version of julia (91c1af8d9a211619). I updated the title accordingly.

**Comment by timholy** [comment]
Awesome, thanks so much for testing this!

I'm not against someone fixing this on the `v0.6` branch, but I won't be doing that myself. Once #57 merges, no one will want 0.6 anyway 😄.

**Comment by timholy** [comment]
I am pretty sure this was fixed by the combination of #133, #144, #145. If this is not working, please see https://timholy.github.io/Revise.jl/latest/config.html#Fixing-a-broken-or-partially-working-installation-1.
